src/averageAges.js

```diff
@@ -37,7 +49,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenWithChild = people.filter(person => withChildren
+    ? people.some(enotherPerson => person.name === enotherPerson.mother)
+    : person.sex === 'f'
+  );
+
+  const allAges = womenWithChild.map(person => person.died - person.born);
```

Comment:
you can make it inside reduce, try without map

Reply:
done ✅